### ISSUE TITLE
Backup encryption passphrase is exposed to dom0 process list

### ISSUE THREAD

**Issue by mig5** [QubesOS/qubes-issues#1582]
```
# ps aux | grep openssl | grep -v grep

miguel    5434  0.2  0.0  40560  4628 ?        S    08:03   0:00 openssl enc -e -aes-256-cbc -pass pass:this should not be seen
miguel    5436  0.0  0.0  40552  4604 ?        S    08:03   0:00 openssl dgst -SHA512 -hmac this should not be seen
```

Although user has a bigger problem if a malicious actor is watching the dom0 process list with ps or top or similar, I think it would be better to use a file descriptor (is this useful example? https://gist.github.com/morgant/9220139)

Maybe also related to #1523 and #971

**Comment by marmarek** [comment]
While `openssl enc` can be easily fixed, `openssl dgst` doesn't support alternative passphrase sources.

**Comment by marmarek** [comment]
Fixed in Qubes 4.0 (thanks to `scrypt` instead of `openssl` usage).
